src/data/finset/sum.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.card
+import data.multiset.sum
+/-!
+# Disjoint sum of finsets
+
+This file defines the disjoint sum of two finsets as `finset (α ⊕ β)`. Beware not to confuse with
+the `finset.sum` operation which computes the additive sum.
+
+## Main declarations
+
+* `finset.disj_sum`: `s.disj_sum t` is the disjoint sum of `s` and `t`.
+-/
+
+open function multiset sum
+
+namespace finset
+variables {α β : Type*} (s : finset α) (t : finset β)
+
+/-- Disjoint sum of finsets. -/
+def disj_sum : finset (α ⊕ β) := ⟨s.1.disj_sum t.1, s.2.disj_sum t.2⟩
+
+@[simp] lemma val_disj_sum : (s.disj_sum t).1 = s.1.disj_sum t.1 := rfl
+
+@[simp] lemma empty_disj_sum : (∅ : finset α).disj_sum t = t.map embedding.inr :=
+val_inj.1 $ multiset.zero_disj_sum _
+
+@[simp] lemma disj_sum_empty : s.disj_sum (∅ : finset β) = s.map embedding.inl :=
+val_inj.1 $ multiset.disj_sum_zero _
+
+@[simp] lemma card_disj_sum : (s.disj_sum t).card = s.card + t.card := multiset.card_disj_sum _ _
+
+variables {s t} {s₁ s₂ : finset α} {t₁ t₂ : finset β} {a : α} {b : β} {x : α ⊕ β}
+
+lemma mem_disj_sum : x ∈ s.disj_sum t ↔ (∃ a, a ∈ s ∧ inl a = x) ∨ ∃ b, b ∈ t ∧ inr b = x :=
+multiset.mem_disj_sum
+
+@[simp] lemma inl_mem_disj_sum : inl a ∈ s.disj_sum t ↔ a ∈ s := inl_mem_disj_sum
+@[simp] lemma inr_mem_disj_sum : inr b ∈ s.disj_sum t ↔ b ∈ t := inr_mem_disj_sum
+
+lemma disj_sum_mono (hs : s₁ ⊆ s₂) (ht : t₁ ⊆ t₂) : s₁.disj_sum t₁ ⊆ s₂.disj_sum t₂ :=
+val_le_iff.1 $ disj_sum_mono (val_le_iff.2 hs) (val_le_iff.2 ht)
+
+lemma disj_sum_mono_left (hs : s₁ ⊆ s₂) (t : finset β) : s₁.disj_sum t ⊆ s₂.disj_sum t :=
+disj_sum_mono hs subset.rfl
```

Comment:
I think this might be worth expressing as `monotone` since the one below is too.
```suggestion
lemma disj_sum_mono_left (t : finset β) : monotone (λ s, s.disj_sum t) :=
λ _ _ hs, disj_sum_mono hs subset.rfl
```